packages/fx-core/src/common/deps-checker/internal/dotnetChecker.ts

```diff
@@ -0,0 +1,534 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as os from "os";
+import * as fs from "fs-extra";
+import * as path from "path";
+import * as child_process from "child_process";
+import * as util from "util";
+import { ConfigFolderName } from "@microsoft/teamsfx-api";
+import { performance } from "perf_hooks";
+
+import { dotnetFailToInstallHelpLink, dotnetExplanationHelpLink } from "../constant/helpLink";
+import { DepsCheckerError } from "../depsError";
+import { runWithProgressIndicator } from "../util/progressIndicator";
+import { cpUtils } from "../util/cpUtils";
+import { isLinux, isWindows } from "../util/system";
+import { DepsCheckerEvent, TelemtryMessages } from "../constant/telemetry";
+import { DepsLogger } from "../depsLogger";
+import { DepsTelemetry } from "../depsTelemetry";
+import { DepsInfo, DepsChecker } from "../depsChecker";
+import { Messages } from "../constant/message";
+
+const execFile = util.promisify(child_process.execFile);
+
+export enum DotnetVersion {
+  v21 = "2.1",
+  v31 = "3.1",
+  v50 = "5.0",
```

Comment:
Make sure you are moving the latest code. E.g., we already have .NET 6.0 support.

Reply:
ok